### ISSUE TITLE
MID$ and LEFT$ clutter variables in stack frame

### ISSUE THREAD

**Issue by orlof** [neilsf/xc-basic3#186]
```
SUB test(x AS INT, y AS INT)
    PRINT x, y
    DIM result AS STRING * 1
    rem result = MID$("nnnn", 0, 1)
    result = LEFT$("nnnn", 1)
    print x, y
END SUB

CALL test(5, 5)
```
<img width="658" alt="image" src="https://user-images.githubusercontent.com/1869202/174852849-a71a3567-45d2-41d9-877c-f3e44aebf78e.png">

**Comment by orlof** [comment]
Related question: I tried to work around this by using peek to get the first character, but I realized that the value of `address of`-operator is relative to current stack frame for local dynamic variables - just out of curiosity, why is that?

**Comment by neilsf** [comment]
Definitely a bug that needs to be fixed. Actually, it's not the local vars that get messed up but the frame pointer itself - so the function's stack frame is read from somewhere else :-D

> the value of `address of`-operator is relative to current stack frame for local dynamic variables - just out of curiosity, why is that?

I don't know, I think I was just too lazy to implement a quick routine that calculates the absolute address of a local variable :-) I can fix this, too, but only in v3.1 because this will be a backwards-incompatible change.
